src/codegen/irgen/irgenerator.cpp

```diff
@@ -487,24 +490,12 @@ class IRGeneratorImpl : public IRGenerator {
                 return rtn;
             }
             case AST_LangPrimitive::GET_ITER: {
-                // TODO if this is a type that has an __iter__, we could do way better than this, both in terms of
-                // function call overhead and resulting type information, if we went with that instead of the generic
-                // version.
-                // (ie we can inline getPystonIter here, whether mechanically with LLVM [would require adding more
-                // optimization passes to make it fast] or by-hand)
-                //
-                // TODO Move this behavior into to the type-specific section (compvars.cpp)?
-                emitter.getBuilder();
                 assert(node->args.size() == 1);
                 CompilerVariable* obj = evalExpr(node->args[0], unw_info);
-
-                ConcreteCompilerVariable* converted_obj = obj->makeConverted(emitter, obj->getBoxType());
+                auto rtn = obj->getiter(emitter, getOpInfoForNode(node, unw_info));
                 obj->decvref(emitter);
-
-                llvm::Value* v = emitter.createCall(unw_info, g.funcs.getPystonIter, { converted_obj->getValue() });
-                assert(v->getType() == g.llvm_value_type_ptr);
-
-                return new ConcreteCompilerVariable(UNKNOWN, v, true);
+                curblock = emitter.getBuilder()->GetInsertBlock();
```

Comment:
Hmm so this call worries me because if we wanted to go with this approach (copy the curblock from emitter to curblock), we would need this line a lot more places.

I think the (hacky) current way of trying to address this is that the emitter has a reference to the curblock, so we should use that.  It looks like it only uses it internally, but we could expose that through some sort of appropriate API, and try to not have multiple places that can get out of sync.